File: Arcade/Intro/57_fileNaming.js
function fileNaming(names) {
    var renamed = [];
    var copies = [];
    
    var goodName = function (file) {
        var x = renamed.indexOf(file);
        if (x < 0) {
            return file;
        } else {
            while (renamed.indexOf(file + "(" + (copies[x]+1) + ")") >= 0) {
                copies[x]++;
            } return file + "(" + (copies[x]+1) + ")";
        }
    }
    for (var i = 0; i < names.length; i++) {
        var n = goodName(names[i]);
        renamed.push(n);
        copies.push(0);
    }
    
    return renamed;
}

File: Arcade/Intro/C++/57_fileNaming.cpp
std::vector<std::string> fileNaming(std::vector<std::string> names) {
    map<string, int> count;
    vector<string> result;
    for(string s : names) {
        string temp=s;
        if(count[s]>0) {
            int i=count[s]-1;
            do {
                i++;
                temp=s;
                temp+="("+to_string(i)+")";
            } while(count[temp]>0);
            count[temp]++;
        }
        count[s]++;
        result.push_back(temp);
    }
    return result;
}
